### DISCUSSION TITLE
Setting acceleration?

### DISCUSSION THREAD

**Discussion by awiswasi** [cyberbotics/webots#6386] [General]
Hi guys,

Is it possible to set the acceleration to a specific XYZ value?
I have a scene with a stationary vehicle. I want to set it's acceleration to, for example, X: 5.0, Y: 0.0, Z: 1.0. Is that possible?

Thanks!

**Comment by omichel** [comment]
You could probably use [wb_supervisor_node_add_force](https://cyberbotics.com/doc/reference/supervisor#wb_supervisor_node_add_force) to achieve this, or use a [Propeller](https://cyberbotics.com/doc/reference/propeller) device mounted on a pan-tilt motor system.

**Comment by awiswasi** [comment]
@omichel Thank you for the suggestion!
I used [wb_supervisor_node_add_force]( https://cyberbotics.com/doc/reference/supervisor#wb_supervisor_node_add_force) and it kind of did what I wanted, however the car seems to float after force is added as if there's no gravity.

My end goal is to feed the simulator XYZ values for acceleration (accelerometer) and rotation (gyro) and kind of visualize what would happen to the car during each set of values. Maybe that clears up my question. Thanks!

**Comment by omichel** [comment]
In that case, you may want to give a try to the other approach I suggested with the propeller.

**Reply by awiswasi** [reply]
Where would I find a propeller and a pan-tilt motor?
The docs aren't that clear about that.

**Reply by omichel** [reply]
Follow the link I posted earlier to find the `Propeller` node. The pan-tilt system can but build with a [Hinge2Joint](https://cyberbotics.com/doc/reference/hinge2joint) and two [RotationalMotor](https://cyberbotics.com/doc/reference/rotationalmotor) nodes.

**Comment by awiswasi** [comment]
@omichel Thank you. I was able to set up a pan-tilt motor system (I think). Here is my object tree:
![Webots](https://github.com/cyberbotics/webots/assets/55606355/a95281f3-048d-4fc6-89ab-efaaf00740c8)

As for my controller code, I tried to set the acceleration to 100 as a test.

```
// Globals
bool has_motor1 = false;
WbDeviceTag motor1;

if (strcmp(name, "motor1") == 0) {
    has_motor1 = true;
}

// Part of my main function
if (has_motor1) {
    motor1 = wb_robot_get_device("motor1");
}

while (wbu_driver_step(64) != -1) {
    if (has_motor1) {
      printf("Success!\n");
      double acceleration = 100.0;
      wb_motor_set_acceleration(motor1, acceleration);
    
    }
}
// ... etc
wbu_driver_cleanup();
```

When I run the simulation, I see that the motor1's acceleration changing to 100, but the car does not move.
What am I doing wrong here? How could I set it to (x = 2.948, y = 1.387, z = 2.006) and see it move like that?

Thank you so much.

**Reply by omichel** [reply]
Something looks wrong: you put an Accelerometer in the endPoint of the Hinge2Joint, whereas you should have put a Propeller node there. Also, you should not set the acceleration of motor1 and motor2, but their position to orientate the propeller in the desired direction before sending a velocity command to the motor of the propeller.
